packages/ui/src/components/Checkbox/index.tsx

```diff
@@ -258,61 +318,84 @@ export const Checkbox = forwardRef(
-          />
-          {!progress ? (
-            <StyledIcon size={size} viewBox="0 0 24 24">
-              <CheckboxIconContainer>
-                <CheckMixedMark x="8" y="11" rx="1" width="8" height="2" />
-                <CheckMark x="8" y="8" rx="1" width="8" height="8" />
-              </CheckboxIconContainer>
-            </StyledIcon>
+      <CheckboxContainer
+        className={className}
+        aria-disabled={disabled}
+        data-visibility={dataVisibility}
+        data-checked={state}
+        data-error={!!error}
+        data-testid={dataTestId}
+      >
+        {progress ? (
+          <StyledActivityContainer>
+            <Loader active size={size} />
+          </StyledActivityContainer>
+        ) : null}
+        <CheckboxInput
+          id={computedName}
+          type="checkbox"
+          aria-invalid={!!error}
+          aria-describedby={error ? `${computedName}-hint` : undefined}
+          aria-checked={state === 'indeterminate' ? 'mixed' : state}
+          aria-label={ariaLabel}
+          checked={state === 'indeterminate' ? false : state}
+          size={size}
+          onChange={onLocalChange}
+          onKeyDown={onKeyDown}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          disabled={disabled}
+          value={value}
+          name={computedName}
+          autoFocus={autoFocus}
+          ref={ref}
+          required={required}
+        />
+
+        {!progress ? (
+          <StyledIcon size={size} viewBox="0 0 24 24">
+            <CheckboxIconContainer>
+              {state !== 'indeterminate' ? (
+                <path
+                  fillRule="evenodd"
+                  clipRule="evenodd"
+                  width={14}
+                  height={14}
+                  d="M15.6678 5.26709C16.0849 5.6463 16.113 6.28907 15.7307 6.70276L9.29172 13.6705C9.10291 13.8748 8.83818 13.9937 8.55884 13.9998C8.2795 14.0058 8.0098 13.8984 7.81223 13.7024L4.30004 10.2185C3.89999 9.82169 3.89999 9.17831 4.30004 8.78149C4.70009 8.38467 5.34869 8.38467 5.74874 8.78149L8.50441 11.5149L14.2205 5.32951C14.6028 4.91583 15.2508 4.88788 15.6678 5.26709Z"
+                  fill="white"
+                />
+              ) : (
+                <CheckMixedMark x="6" y="11" rx="1" width="12" height="2" />
+              )}
+            </CheckboxIconContainer>
+          </StyledIcon>
+        ) : null}
+
+        <Stack gap={0.25}>
+          <Row templateColumns="11fr 1fr" alignItems="center">
+            {children ? (
+              <StyledLabel htmlFor={computedName}>{children}</StyledLabel>
+            ) : null}
+            {required ? (
+              <sup>
+                <Icon name="asterisk" size={10} color="danger" />
+              </sup>
+            ) : null}
+          </Row>
+
+          {helper ? (
+            <Text variant="bodySmall" as="p" prominence="weak" color="neutral">
+              {helper}
+            </Text>
           ) : null}
```

Comment:
mb we should also change the component to allow content to be well aligned with the label ? we may need a dedicated prop for the `label` and use children as optional content ?

Reply:
Sorry, not sure to understand well
you want that we use label as label props and use the children’s props for optional content ? what is the matter ?